bin/varnishd/cache/cache_conn_pool.c

```diff
@@ -249,17 +259,39 @@ VCP_Rel(struct conn_pool **cpp)
 		(void)shutdown(pfd->fd, SHUT_RDWR);
 		cp->n_kill++;
 	}
-	while (cp->n_kill) {
-		Lck_Unlock(&cp->mtx);
-		(void)usleep(20000);
+
+	ALLOC_OBJ(dcp, DYING_CP_MAGIC);
```

Comment:
`AN(dcp);` ? Do we really need to allocate it? Why not reuse `cp`  ? Currently it does not belong to any tree here, either use that, or add a list member to `cp` which can be used here instead.

Reply:
Marking this as duplicate. See below.